src/org/opensolaris/opengrok/search/context/QueryMatchers.java

```diff
@@ -89,9 +90,19 @@ private void getTerms(Query query) {
             getTerm((TermQuery) query);
         } else if (query instanceof PrefixQuery) {
             getPrefix((PrefixQuery) query);
+        } else if (query instanceof RegexpQuery) {
```

Comment:
well I wasn't sure if Summarizer is used for the short listing, or for the longer listing when there is an ALL link in results
but it IS definitely used and based on your observation it seems to be the long listing
(so test is to create file which matches on more than 10 lines, then you get an ALL link from search results page for that doc hit ... )